src/remote_worker.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2019 dcurl Developers.
+ * Use of this source code is governed by MIT license that can be
+ * found in the LICENSE file.
+ */
+
+#include "constants.h"
+#include "dcurl.h"
+#include "remote_common.h"
+
+int main(int argc, char const *const *argv)
+{
+    char trytes[TRANSACTION_TRYTES_LENGTH];
+    char buf[4];
+    int mwm;
+
+    amqp_connection_state_t conn;
+    amqp_envelope_t envelope;
+
+    dcurl_init();
+
+    if (!connect_broker(&conn, 1))
+        goto fail_to_exit;
+
+    if (!set_consuming_queue(&conn, 1, "incoming_queue"))
+        goto fail_to_exit;
+
+    for (;;) {
+        if (!consume_message(&conn, 1, &envelope))
+            goto fail_to_exit;
+
+#if defined(ENABLE_DEBUG)
+        printf(MSG_PREFIX "Delivery %u, exchange %.*s, routingkey %.*s, callback queue: %s "
+            "\n",
+            (unsigned) envelope.delivery_tag, (int) envelope.exchange.len,
+            (char *) envelope.exchange.bytes, (int) envelope.routing_key.len,
+            (char *) envelope.routing_key.bytes,
+            (char *) envelope.message.properties.reply_to.bytes);
+        if (envelope.message.properties._flags & AMQP_BASIC_CONTENT_TYPE_FLAG) {
+            printf(MSG_PREFIX "Content-type: %.*s\n",
+                   (int) envelope.message.properties.content_type.len,
+                   (char *) envelope.message.properties.content_type.bytes);
+        }
+
+#endif
+
+        /* Message body format: transacton | mwm */
+        memcpy(trytes, envelope.message.body.bytes, TRANSACTION_TRYTES_LENGTH);
+        memcpy(buf, envelope.message.body.bytes + TRANSACTION_TRYTES_LENGTH, 4);
+        sscanf(buf, "%d", &mwm);
```

Comment:
The reason why sscanf might be considered bad is because it does not require you to specify maximum string width for string arguments, which could result in overflows if the input read from the source string is longer. 

Reference: https://stackoverflow.com/questions/9245682/in-c-what-is-a-safe-alternative-to-sscanf/16043903